### PR TITLE
Refactor plugin install process

### PR DESCRIPTION
**Pull request by nhunzaker**
This commit uses a more straightforward queue when installing plugins and removes the need to add a `__start` method to plugins.

### CODE REVIEW

File: src/plugin.js [hunk 1]
```diff
@@ -7,16 +7,25 @@
...
+  return function (error) {
+    // Halt execution of all future plugin installation if there is an error
+    if (error) {
+      return next(error)
+    }
 
-  register(app, options, next) {
-    next(null)
+    // Plugins might not have a register method. In this case, just continue through
+    return plugin.register ? plugin.register(plugin.app, plugin.options, next) : next(null)
```
Comment (solomonhawk) [inline src/plugin.js]: Check for `typeof === 'function'` perhaps?

**Comment by solomonhawk** [comment]
:+1:

File: src/plugin.js (hunk 1 above)
Comment (nhunzaker) [inline src/plugin.js]: Hmmm. That would kind of be a silent error though. What about a dev-only error?

**Comment by jeremyfrank** [comment]
:+1:

File: src/plugin.js (hunk 1 above)
Comment (nhunzaker) [inline src/plugin.js]: Thoughts on ca2747cd5c144af35049f4432d193568b3929199?

Comment (solomonhawk) [inline src/plugin.js]: Looks good

**Pull request merged**